.github/workflows/pr-snapshot.yaml

```diff
@@ -18,35 +20,34 @@ jobs:
         run: |
           pr_number="$(head -n 2 /tmp/artifacts/metadata.txt | tail -n 1)"
           pr_commit="$(tail -n 1 /tmp/artifacts/metadata.txt)"
-          echo PR_COMMIT="$pr_commit" >> "$GITHUB_OUTPUT"
-          echo PR_NUMBER="$pr_number" >> "$GITHUB_OUTPUT"
+          echo PR_COMMIT="$pr_commit" >> "GITHUB_ENV"
+          echo PR_NUMBER="$pr_number" >> "GITHUB_ENV"
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
-          ref: ${{ steps.metadata.outputs.PR_COMMIT }}
+          ref: ${{ env.PR_COMMIT }}
       - uses: ./.github/actions/commit-status/start
         with:
           name: "${{ github.workflow }} / ${{ github.job }} (pull_request_review)"
-          git_ref: ${{ steps.metadata.outputs.PR_COMMIT }}
+          git_ref: ${{ env.PR_COMMIT }}
       - uses: ./.github/actions/install-deps
       - uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
         with:
           role-to-assume: 'arn:aws:iam::${{ vars.ECR_ACCOUNT_ID }}:role/${{ vars.ECR_SNAPSHOT_ROLE_NAME }}'
           aws-region: ${{ vars.ECR_REGION }}
       - run: make snapshot
         env:
-          GH_PR_NUMBER: ${{steps.metadata.outputs.PR_NUMBER}}
+          GH_PR_NUMBER: ${{ env.PR_NUMBER }}
       - uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
         with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
           script: |
             github.rest.issues.createComment({
-              issue_number: `${{steps.metadata.outputs.PR_NUMBER}}`,
+              issue_number: process.env.PR_NUMBER,
```

Comment:
Why does this one use a non bracketed reference, but the others referencing PR_NUMBER do? Seems like one of these is wrong, or all of them should refer to process.env.PR_NUMBER.

Reply:
I can move this up for clarity. Repository environment variables aren't user-controller, but I can also see how this might be confusing for someone just reading through the script files here.